pkg/plugin/plugin_canary.go

```diff
@@ -58,3 +60,42 @@ func (r *RpcPlugin) newCanaryDest(stableDest *solov2.DestinationReference, rollo
 	newDest.GetRef().Name = rollout.Spec.Strategy.Canary.CanaryService
 	return newDest, nil
 }
+
+func (r *RpcPlugin) handleHeaderRoute(ctx context.Context, routeTables []*GlooMatchedRouteTable, headerRoute *GlooMatchedHttpRoutes) pluginTypes.RpcError {
+	var combinedError error
+	for _, rt := range routeTables {
+		originalRouteTable := &networkv2.RouteTable{}
+		rt.RouteTable.DeepCopyInto(originalRouteTable)
+
+		newRoutes := make([]*networkv2.HTTPRoute, 1)
+		newRoutes[0] = headerRoute.HttpRoute
+		newRoutes = append(newRoutes, rt.RouteTable.Spec.Http...)
+		rt.RouteTable.Spec.Http = newRoutes
+
+		if !r.IsTest {
```

Comment:
I know this existed prior to your contribution, but implementations with testing specific functionality is a smell. I think we're doing this because of a dependency on k8s client. Would you be able to mock this instead? I think it would go a long way to cleaning up these functions.

Reply:
yeah, super agree. We should file this as an issue imo. improved test mocks/e2e tests are orthogonal to setHeaderRoute existing. WDYT?

Reply:
Resolved!